src/platform/packages/shared/kbn-object-utils/src/calculate_object_diff.test.ts

```diff
@@ -40,4 +40,34 @@ describe('calculateObjectDiff', () => {
     expect(removed).toEqual({});
     expect(updated).toEqual({});
   });
+
+  it('should handle array fields correctly', () => {
+    const { added, removed, updated } = calculateObjectDiff(
+      { alpha: [1, 2, 3], beta: [4, 5, 6] },
+      { alpha: [1, 2, 3], beta: [4, 5, 7] }
+    );
+    expect(added).toEqual({});
+    expect(removed).toEqual({});
+    expect(updated).toEqual({ beta: [undefined, undefined, 7] });
```

Comment:
silly nit: this algo / API cannot communicate when/if the value was `updated` to `undefined` at index 0 or 1.